x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/_mocks_/mocked_engines.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FetchEnginesApiLogicResponse } from '../../../api/engines/fetch_engines_api_logic';
```

Comment:
nit: This creates a circular dependency, it may make more sense to just defined the `mockedEngines` list in `fetch_engines_api_logic` files.

webpack handles the circular dependency so this isn't a must change. 

Another option would be to import `EngineListDetails` and use `EngineListDetails[]` directly instead of `FetchEnginesApiLogicResponse`